src/formattedcode/output_xml.py

```diff
@@ -0,0 +1,54 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# ScanCode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/nexB/scancode-toolkit for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+import json 
+
+from json2xml import json2xml
+from json2xml.utils import readfromstring
+from formattedcode import FileOptionType
+from formattedcode import output_json
+from commoncode.cliutils import PluggableCommandLineOption
+from commoncode.cliutils import OUTPUT_GROUP
+from plugincode.output import output_impl
+from plugincode.output import OutputPlugin
+
+"""
+Output plugin to write scan results as XML.
+"""
+
+@output_impl
+class XmlOutput(OutputPlugin):
+
+    options = [
+        PluggableCommandLineOption(('--xml', 'output_xml',),
+            type=FileOptionType(mode='w', encoding='utf-8', lazy=True),
+            metavar='FILE',
+            help='Write scan output as XML to FILE.',
+            help_group=OUTPUT_GROUP,
+            sort_order=20
+        ),
+    ]
+
+    def is_enabled(self, output_xml, **kwargs):
+        return output_xml
+
+    def process_codebase(self, codebase, output_xml, **kwargs):
+        results = output_json.get_results(codebase, as_list=True, **kwargs)
+        results = json.dumps(results)
```

Comment:
This is likely problematic: you dump JSON to a string that this further parsed to be converted to XML?

Reply:
Yes, so far it has given proper reports for a variety of cases.

Reply:
> Yes, so far it has given proper reports for a variety of cases.

I get that, but you have structured data in `results` ... this could be converted directly to XML ... no need to dump as jSON and reparse. You should research options in https://github.com/vinitkumar/json2xml/blob/1597080ad93302837c9e0855ccdfe1c0e204d39b/json2xml/dicttoxml.py#L388

That said, I wonder what's the benefit of having an XML format over JSON overall?

Reply:
Ok I will check with that. As for any benefit, I feel JSON is still the best. I wanted to add that functionality to the application and contribute so I worked on it. Thank you for suggesting that.